qutebrowser/misc/keyhintwidget.py

```diff
@@ -90,19 +83,29 @@ def update_keyhint(self, modename, prefix):
         Args:
             prefix: The current partial keystring.
         """
-        if not prefix or not self._enabled:
+        if not prefix:
+            self._show_timer.stop()
             self.hide()
             return
 
+        blacklist = config.get('ui', 'keyhint-blacklist') or []
         keyconf = objreg.get('key-config')
+
+        def blacklisted(keychain):
+            return any([glob for glob in blacklist
+                        if fnmatch.fnmatchcase(keychain, glob)])
```

Comment:
`any()` already checks if all arguments are true, and you can use a [generator expression](https://docs.python.org/3.5/reference/expressions.html#grammar-token-generator_expression).

In other words, you can simplify this to `any(fnmatch.fnmatchcase(keychain, glob) for glob in blacklist)`.